src/Bicep.Core/Modules/ModuleReference.cs

```diff
@@ -40,5 +40,10 @@ protected ModuleReference(string scheme, Uri parentModuleUri)
         /// Gets a value indicating whether this reference points to an external module.
         /// </summary>
         public abstract bool IsExternal { get; }
+
+        /// <summary>
+        /// Gets or sets documentation url.
+        /// </summary>
+        public virtual string? DocumentationUrl { get; set; }
```

Comment:
>DocumentationUrl [](http://example.com/codeflow?start=31&length=16)

The intent of the `ModuleReference` class is to capture everything that can be parsed out from a module reference string (everything after `br:` basically). Documentation URI is not part of that string.

Reply:
I have an idea how to avoid passing this info around in the module ref. 

The OCI manifest is serialized to disk under `%userprofile%\.bicep\br` for each restored module. I'm wondering if we can have the language server retrieve the module manifest from the cache on demand during a hover on a module instead.

The logic to calculate the path is in `OciModuleRegistry.GetModuleDirectoryPath`. You could potentially add a new method to `IModuleRegistry` to try to retrieve documentation URI (won't be possible for local or template spec modules) using that method.

Reply:
The key part is to ensure that the abstraction is preserved, so the language server shouldn't need to know where the modules are cached, but just ask for docs for a module.

Reply:
Thanks for the explanation. The DocumentationUrl in ModuleReference was unused. I removed that and also refactored the code a bit here:
https://github.com/Azure/bicep/commit/2918b4e4e87f1b4dc02e8bff93de298e6590652b
https://github.com/Azure/bicep/commit/9f8e87c0f94227ed14f4969b0d363cbec19f70f6